### PR TITLE
[LLM Runtime]Magicoder graph

### PR DESCRIPTION
**Pull request by intellinjun**
## Type of Change

feature or bug fix or documentation or others
API changed or not
not

## Description
- change convert_llama.py  and ne_layers.c for magicoder
- new parameter rope_scaling
detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c [hunk 1]
```diff
@@ -7867,9 +7874,8 @@ static void ne_compute_forward_rope_f32(const struct ne_compute_params* params,
   NE_ASSERT(src1->type == NE_TYPE_I32);
   NE_ASSERT(ne_nelements(src1) == 5 + bs);  // 5 + bs params
 
-  float freq_base = 10000.0f;
-  memcpy(&freq_base, dst->op_params, sizeof(float));
-  static const float freq_scale = 1.0f;
+  const float freq_base = ((float*)(dst->op_params))[0];
+  const float freq_scale = 1 / ((float*)(dst->op_params))[1];
```
Comment (DDEle) [inline intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c:7878]: Change correspondingly in `ne_compute_forward_rope_f16` and  `kv_cache_init` (`models/model_utils/model_utils.cpp`)

Comment (a32543254) [reply intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c:7824]: Fixed, thanks for your comment

**Review by zhenwei-intel — APPROVED** [review]

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by airMeng** [comment]
any comprehensive expansion test？

**Comment by a32543254** [comment]
> any comprehensive expansion test？

This model share same structure of llama model, so we believe the llama2 model extension test should be ok for us.

**Comment by airMeng** [comment]
> This model share same structure of llama model, so we believe the llama2 model extension test should be ok for us.

then at least an extension test on magiccoder weight is necessary then we can check whether the output is reasonable?

**Comment by a32543254** [comment]
> > This model share same structure of llama model, so we believe the llama2 model extension test should be ok for us.
> 
> then at least an extension test on magiccoder weight is necessary then we can check whether the output is reasonable?

Extension test add here,
https://github.com/intel/intel-extension-for-transformers/pull/1053/commits/4c8680623331f7b06f2ac0dafc1eca6e2fbc146e

**Comment by VincyZhang** [comment]
need to migrate to NeuralSpeed?

**Comment by a32543254** [comment]
> need to migrate to NeuralSpeed?

will keep in ITREX for now. 
Because Lv liang want to use it for demo show at the begin of next month

**Comment by VincyZhang** [comment]
> > need to migrate to NeuralSpeed?
> 
> will keep in ITREX. Because Lv liang want to use it for demo show at the begin of next month

We can keep it in ITREX first, but need to migrate to NeuralSpeed afterwards

**Pull request merged**